libs/wingc/src/parser.rs

```diff
@@ -252,6 +253,17 @@ impl Parser<'_> {
 		})
 	}
 
+	fn build_break_statement(&self, statement_node: &Node) -> DiagnosticResult<StmtKind> {
+		let mut cursor = statement_node.walk();
+		while let Some(parent) = cursor.node().parent() {
```

Comment:
for some reason, `cursor.goto_parent()` is always `false`, walking around with this ugly approach. LMK what did I miss

Reply:
After playing around with this `goto_parent` API, it works only for the cursor previously went through, e.g., in this use case, it returns `true` IIF we do `cursor.goto_first_child()` first. In other word, it considers the `walk()` TreeNode as the root of the tree. However, I cannot find any doc about this behavior, LMK

Reply:
This makes a lot of sense, let me take a look